Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -2040,7 +2040,7 @@ void didReceiveData(const WebCore::SharedBuffer& buffer)
     {
         if (!m_connection)
             return;
-        buffer.forEachSegment([&](auto segment) {
+        buffer.forEachSegment([this, protectedThis = Ref { *this }](auto segment) {
```

Comment:
I think we just need to mark the `forEachSegment()` parameter as `NOESCAPE` and we won't need this change. This lambda runs synchronously so it is safe.

Reply:
IIUC this is a change to `SharedBuffer::forEachSegment()` and probably also SharedBuffer::forEachSegmentAsSharedBuffer()`, would you prefer if I make that a different patch? I think it would make sense.

Reply:
Done in https://github.com/WebKit/WebKit/pull/39813